synmark/__main__.py

```diff
@@ -12,54 +12,60 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 import sys
-from argparse import REMAINDER
+from argparse import REMAINDER, Namespace
 from contextlib import redirect_stderr
 from io import StringIO
+from typing import Any, Callable, Coroutine, List, TypeVar, cast
 
 import pyperf
 
 from twisted.internet.defer import Deferred, ensureDeferred
 from twisted.logger import globalLogBeginner, textFileLogObserver
 from twisted.python.failure import Failure
 
+from synapse.types import ISynapseReactor
 from synmark import make_reactor
 from synmark.suites import SUITES
 
 from tests.utils import setupdb
 
+T = TypeVar("T")
 
-def make_test(main):
+
+def make_test(
+    main: Callable[[ISynapseReactor, int], Coroutine[float, Any, Any]]
+) -> Callable[[int], float]:
     """
     Take a benchmark function and wrap it in a reactor start and stop.
     """
 
-    def _main(loops):
+    def _main(loops: int) -> float:
         reactor = make_reactor()
 
         file_out = StringIO()
         with redirect_stderr(file_out):
-            d = Deferred()
-            d.addCallback(lambda _: ensureDeferred(main(reactor, loops)))
+            d: "Deferred[float]" = Deferred()
+            d.addCallback(lambda _: ensureDeferred(main(reactor, loops)))  # type: ignore
```

Comment:
What's the ignore for here? Can we ignore a specific error code only?

Reply:
Arg, it is `ensureDeferred` complaining about getting a coroutine, I think? Will double check.